src/linear_algebra/affine_space.lean

```diff
@@ -1231,6 +1368,74 @@ end
 
 end affine_space
 
+section affine_independent
+
+open affine_space
+
+variables {k : Type*} {V : Type*} {P : Type*} [ring k] [add_comm_group V] [module k V]
+variables [affine_space k V P] {ι : Type*}
+
+/-- If a family is affinely independent, the spans of points indexed
+by disjoint subsets of the index type are disjoint, if the underlying
+ring is nontrivial. -/
+lemma affine_span_disjoint_of_disjoint_of_affine_independent [nontrivial k] {p : ι → P}
+    (ha : affine_independent k V p) {s1 s2 : set ι} (hd : s1 ∩ s2 = ∅) :
+  (affine_span k V (p '' s1) : set P) ∩ affine_span k V (p '' s2) = ∅ :=
+begin
+  by_contradiction hne,
+  change (affine_span k V (p '' s1) : set P) ∩ affine_span k V (p '' s2) ≠ ∅ at hne,
+  rw set.ne_empty_iff_nonempty at hne,
+  rcases hne with ⟨p0, hp0s1, hp0s2⟩,
+  rw [set.image_eq_range, affine_subspace.mem_coe] at hp0s1 hp0s2,
+  rw [mem_affine_span_iff_eq_affine_combination,
+      ←finset.eq_affine_combination_subset_iff_eq_affine_combination_subtype] at hp0s1 hp0s2,
+  rcases hp0s1 with ⟨fs1, hfs1, w1, hw1, hp0s1⟩,
+  rcases hp0s2 with ⟨fs2, hfs2, w2, hw2, hp0s2⟩,
+  rw finset.affine_combination_indicator_subset _ _ _ (finset.subset_union_left fs1 fs2)
+    at hp0s1,
+  rw finset.affine_combination_indicator_subset _ _ _ (finset.subset_union_right fs1 fs2)
+    at hp0s2,
+  let w3 := set.indicator ↑fs1 w1 - set.indicator ↑fs2 w2,
+  have h0 : (fs1 ∪ fs2).weighted_vsub V p w3 = 0,
+  { simp [←finset.affine_combination_vsub, ←hp0s1, ←hp0s2] },
+  have hw3 : ∑ i in fs1 ∪ fs2, w3 i = 0,
+  { change ∑ i in fs1 ∪ fs2, (set.indicator ↑fs1 w1 i - set.indicator ↑fs2 w2 i) = 0,
+    rw [finset.sum_sub_distrib,
+        ←set.sum_indicator_subset _ (finset.subset_union_left fs1 fs2),
+        ←set.sum_indicator_subset _ (finset.subset_union_right fs1 fs2), hw1, hw2, sub_self] },
+  replace ha := ha (fs1 ∪ fs2) w3 hw3 h0,
+  have hnz : ∑ i in fs1, w1 i ≠ 0 := hw1.symm ▸ one_ne_zero,
+  rcases finset.exists_ne_zero_of_sum_ne_zero hnz with ⟨i, hifs1, hin0⟩,
+  change w1 i ≠ 0 at hin0,
+  replace ha := ha i (finset.mem_union_left _ hifs1),
+  change set.indicator ↑fs1 w1 i - set.indicator ↑fs2 w2 i = 0 at ha,
+  rw ←finset.mem_coe at hifs1,
+  rw set.indicator_of_mem hifs1 _ at ha,
+  rw ←set.disjoint_iff_inter_eq_empty at hd,
+  have hifs2 : i ∉ (↑fs2 : set ι) :=
+    λ h, set.not_disjoint_iff.2
+           ⟨i, hifs1, h⟩
+           (set.disjoint_of_subset_right hfs2 (set.disjoint_of_subset_left hfs1 hd)),
+  rw [set.indicator_of_not_mem hifs2 _, sub_zero] at ha,
+  exact hin0 ha
+end
+
```

Comment:
I'd like to see a lemma like:
```lean
@[simp]
lemma mem_affine_span_iff [nontrivial k] {p : ι → P}
    (ha : affine_independent k V p) (i : ι) (s : set ι) :
  p i ∈ affine_span k V (p '' s) ↔ i ∈ s :=
sorry
```

Then `not_mem_affine_span_diff_of_affine_independent` would be a consequence of this plus a bit of set theory.

Reply:
Done (where the "bit of set theory" turns out to be `by simp [ha]`).